lib/connect/index.js

```diff
@@ -21,19 +21,57 @@ protocols.wss = require('./ws')
  *
  * @param {Object} [opts] option object
  */
-function parseAuthOptions (opts) {
+function parseAuthOptions (auth, opts) {
   var matches
-  if (opts.auth) {
-    matches = opts.auth.match(/^(.+):(.+)$/)
+  if (auth) {
+    matches = auth.match(/^(.+):(.+)$/)
     if (matches) {
       opts.username = matches[1]
       opts.password = matches[2]
     } else {
-      opts.username = opts.auth
+      opts.username = auth
     }
   }
 }
 
+function parseBrokerUrl (brokerUrl) {
+  var opts = {}
+  var parsed = url.parse(brokerUrl, /* parseQueryString = */ true)
+  if (parsed.port != null) {
+    opts.port = Number(parsed.port)
+  }
+  if (parsed.protocol) {
+    opts.protocol = parsed.protocol.replace(/:$/, '')
+  }
+
+  // Parse username/password
+  parseAuthOptions(parsed.auth, opts)
+
+  // Support clientId passed in the query string of the url
+  if (parsed.query && typeof parsed.query.clientId === 'string') {
+    opts.clientId = parsed.query.clientId
+  }
+
+  return xtend(
+    opts,
+      // username
+      // password
+      // port
+      // protocol
+      // clientId
+    {
+      // hostname: parsed.hostname,
```

Comment:
You could argue that to be consistent with deprecating, should leave the `hostname` until it's completely removed. Assuming you guys are down with that direction.